### PR TITLE
Add PublicAPI baselines for Middleware

### PR DESCRIPTION
**Pull request by Pilchie**

### CODE REVIEW

File: src/Middleware/CORS/src/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -0,0 +1,118 @@
+#nullable enable
+Microsoft.AspNetCore.Builder.CorsEndpointConventionBuilderExtensions
+Microsoft.AspNetCore.Builder.CorsMiddlewareExtensions
+Microsoft.AspNetCore.Cors.CorsPolicyMetadata
```
Comment (Tratcher) [inline src/Middleware/CORS/src/PublicAPI.Unshipped.txt:4]: How do these Unshipped and Shipped files work? I'd expect Unshipped to be the APIs new to 5.0 and Shipped to be the 3.1 baseline? Most of the APIs in this Unshipped file are not new.

Comment (Pilchie) [reply src/Middleware/CORS/src/PublicAPI.Unshipped.txt:4]: We manually move to the Shipped file once we ship.  Right now I'm just trying to get everything into the unshipped files as a baseline, and then we'll take our 5.0 surface area and move it to shipped at once.

**Review by Tratcher — APPROVED** [review]

**Comment by Pilchie** [comment]
@dougbu - any objections to me merging?

File: src/Middleware/Rewrite/src/IISUrlRewriteOptionsExtensions.cs [hunk 2]
```diff
@@ -21,6 +22,7 @@ public static class IISUrlRewriteOptionsExtensions
         /// <param name="fileProvider">The <see cref="IFileProvider"/> </param>
         /// <param name="filePath">The path to the file containing UrlRewrite rules.</param>
         /// <param name="alwaysUseManagedServerVariables">Server variables are by default sourced from the server if it supports the <see cref="IServerVariablesFeature"/> feature. Use <c>true</c> to disable that behavior</param>
+        [SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters", Justification = "<Pending>")]
```
Comment (dougbu) [inline src/Middleware/Rewrite/src/IISUrlRewriteOptionsExtensions.cs:25]: Either here or in a follow up, please fill in the `<Pending>` bit

**Review by dougbu — APPROVED** [review]

**Comment by dougbu** [comment]
> any objections to me merging?

Not really though we shouldn't leave `<Pending>` justifications around for long

File: src/Middleware/Rewrite/src/IISUrlRewriteOptionsExtensions.cs (hunk 2 above)
Comment (Pilchie) [reply src/Middleware/Rewrite/src/IISUrlRewriteOptionsExtensions.cs:25]: Whoops - good catch.

**Pull request merged**